core/trino-main/src/main/java/io/trino/operator/DirectExchangeClient.java

```diff
@@ -369,6 +374,7 @@ private void releaseMemoryContext()
 
     private synchronized void requestComplete(HttpPageBufferClient client)
     {
+        requestDuration.add(client.getLastRequestDuration());
```

Comment:
I think client might start another request asynchroniously due to `requestComplete` being delayed

Reply:
You could add `long requestDurationMillis` to `HttpPageBufferClient.ClientCallback#requestComplete`